c/cb-ls.c

```diff
@@ -0,0 +1,134 @@
+#include <dirent.h>    /* ino_t */
+#include <errno.h>     /* stderr */
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>    /* qsort */
+#include <unistd.h>    /* getopt */
+
+#define LOTS 2000
+
+const int true = 1;
+const int false = 0;
+
+struct cb_dir {
+    ino_t ino;
+    char* name;
+    int hidden;
+} cb_dirs[LOTS];
```

Comment:
Why fixed allocation rather than malloc?

Reply:
@dgriffith I opted for this approach because I made the decision to set a limit ahead of time for how long the array would be. I wasn't sure if taking two passes through the directory would make sense -- once to get a count of files to use to dynamically allocate space, and then a second pass to actually populate the newly-created array. I suppose from the perspective of the user, a slower `ls` would be better than an `ls` that crapped out in medium-large directories.